veggieseasons/lib/main.dart

```diff
@@ -83,14 +94,141 @@ class _VeggieAppState extends State<VeggieApp> with RestorationMixin {
           create: (_) => Preferences()..load(),
         ),
       ],
-      child: CupertinoApp(
+      child: CupertinoApp.router(
         theme: Styles.veggieThemeData,
         debugShowCheckedModeBanner: false,
-        home: const HomeScreen(restorationId: 'home'),
         restorationScopeId: 'app',
+        routerConfig: GoRouter(
+          navigatorKey: _rootNavigatorKey,
+          restorationScopeId: 'router',
+          initialLocation: '/list',
+          redirect: (context, state) {
+            if (state.path == '/') {
+              return '/list';
+            }
+            return null;
+          },
+          debugLogDiagnostics: true,
+          routes: [
+            ShellRoute(
+              navigatorKey: _shellNavigatorKey,
+              pageBuilder: (context, state, child) {
+                return CupertinoPage(
+                  restorationId: 'router.shell',
+                  child: HomeScreen(
+                    restorationId: 'home',
+                    child: child,
+                    onTap: (index) {
+                      if (index == 0) {
+                        context.go('/list');
+                      } else if (index == 1) {
+                        context.go('/favorites');
+                      } else if (index == 2) {
+                        context.go('/search');
+                      } else {
+                        context.go('/settings');
+                      }
+                    },
+                  ),
+                );
+              },
+              routes: [
+                GoRoute(
+                  path: '/list',
+                  pageBuilder: (context, state) {
+                    return FadeTransitionPage(
```

Comment:
I'm not seeing a fade transition when I tap on destinations in the bottom navigation bar - is that what this is supposed to do?

Reply:
True. I have modified the FadeTransitionPage to do the same as the CustomTransitionPage example you linked, and the issue is still there. Do you know if this could be a limitation on using it inside a ShellRoute?